org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/handlers/JDTLanguageServer.java

```diff
@@ -154,6 +158,12 @@ public void initialized(InitializedParams params) {
 	@Override
 	public CompletableFuture<Object> shutdown() {
 		logInfo(">> shutdown");
+		GradleBuildSupport.saveModel();
```

Comment:
so, instead of calling GradleBuildSupport.saveModel() explicitely, I think it'd be better to add ProjectManager as an ISaveParticipant to the workspace.
Then ProjectManager.prepareToSave could call GradleBuildSupport.saveModel();

Reply:
ISaveParticipant is called too often.

Reply:
so you can filter by ISaveContext.getKind() == FULL_SAVE and/or add the ISaveParticipant right before saving the workspace